packages/sit-onyx/src/components/OnyxDataGrid/features/sorting/sorting.ts

```diff
@@ -98,10 +109,17 @@ export const useSorting = <TEntry extends DataGridEntry>(options?: SortOptions<T
 
     return {
       name: SORTING_FEATURE,
-      watch: [sortState, intlCompare],
+      watch: [sortState, i18n.locale],
```

Comment:
```suggestion
      watch: [sortState, collator],
```

Don't we want to watch the collator instead because it could also be passed as ref by the project

Reply:
done